aptos-move/e2e-move-tests/src/tests/scripts.rs

```diff
@@ -80,3 +81,181 @@ fn read_coin(h: &MoveHarness, account: &AccountAddress) -> u64 {
+        ..aptos_framework::BuildOptions::default()
+    };
+
+    let package = aptos_framework::BuiltPackage::build(
+        common::test_dir_path("../../../move-examples/scripts/two_by_two_transfer"),
+        build_options,
+    )
+    .expect("building package must succeed");
+
+    let alice_start = read_coin(&h, alice.address());
+    let bob_start = read_coin(&h, bob.address());
+    let carol_start = read_coin(&h, carol.address());
+    let david_start = read_coin(&h, david.address());
+    let payer_start = read_coin(&h, payer.address());
+
+    let code = package.extract_script_code()[0].clone();
+    let script = Script::new(code, vec![], vec![
+        TransactionArgument::U64(amount_alice),
+        TransactionArgument::U64(amount_bob),
+        TransactionArgument::Address(*carol.address()),
+        TransactionArgument::Address(*david.address()),
+        TransactionArgument::U64(amount_carol),
+    ]);
+
+    let transaction = TransactionBuilder::new(alice.clone())
+        .secondary_signers(vec![bob.clone(), payer.clone()])
+        .script(script)
+        .sequence_number(h.sequence_number(alice.address()) | (1u64 << 63))
+        .max_gas_amount(1_000_000)
+        .gas_unit_price(1)
+        .sign_multi_agent();
+
+    let output = h.executor.execute_transaction(transaction);
+    assert_success!(output.status().to_owned());
+    h.executor.apply_write_set(output.write_set());
+
+    let alice_end = read_coin(&h, alice.address());
+    let bob_end = read_coin(&h, bob.address());
+    let carol_end = read_coin(&h, carol.address());
+    let david_end = read_coin(&h, david.address());
+    let payer_end = read_coin(&h, payer.address());
+
+    // Make sure sender alice doesn't pay gas
+    assert_eq!(alice_start - amount_alice, alice_end);
+    assert_eq!(bob_start - amount_bob, bob_end);
+    assert_eq!(carol_start + amount_carol, carol_end);
+    assert_eq!(david_start + amount_david, david_end);
+    // Make sure payer pays
+    assert_eq!(payer_start - output.gas_used(), payer_end);
+}
+
+#[test]
+fn test_two_to_two_transfer_gas_payer_without_feature() {
+    let mut h = MoveHarness::new();
+
+    let alice = h.new_account_at(AccountAddress::from_hex_literal("0xa11ce").unwrap());
+    let bob = h.new_account_at(AccountAddress::from_hex_literal("0xb0b").unwrap());
+    let carol = h.new_account_at(AccountAddress::from_hex_literal("0xca501").unwrap());
+    let david = h.new_account_at(AccountAddress::from_hex_literal("0xda51d").unwrap());
+    let payer = h.new_account_at(AccountAddress::from_hex_literal("0xea51d").unwrap());
+
+    let amount_alice = 100;
+    let amount_bob = 200;
+    let amount_carol = 50;
+
+    let build_options = aptos_framework::BuildOptions {
+        with_srcs: false,
+        with_abis: false,
+        with_source_maps: false,
+        with_error_map: false,
+        ..aptos_framework::BuildOptions::default()
+    };
+
+    let package = aptos_framework::BuiltPackage::build(
```

Comment:
nit: This looks duplicate. May be refactor this part of constructing a transfer tx with a gas payer out into a separate function?